portal/static/js/initialQueries.js

```diff
@@ -908,72 +742,42 @@
                 self.postDemoData($(this));
             });
         });
+        __convertToNumericField($("#date, #year"));
     };
 
-    FieldsChecker.prototype.rolesGroupEvent = function(fields) {
+    FieldsChecker.prototype.rolesGroupEvent = function() {
         var self = this, tnthAjax = this.__getDependency("tnthAjax");
-        fields.forEach(function(item) {
-            $(item).on("click", function() {
-                var roles = [], theVal = $(this).val();
-                roles.push({name: theVal});
-                tnthAjax.putRoles(self.userId,{"roles": roles}, $("#rolesGroup"));
-                self.handlePostEvent(self.getSectionContainerId($(this)));
-                if (theVal === "patient") {
-                    $("#clinicalContainer").attr("skipped", "false");
-                    $("#orgsContainer").attr("skipped", "false");
-                    $("#date").attr("required", "required").attr("skipped", "false");
-                    $("#month").attr("required", "required").attr("skipped", "false");
-                    $("#year").attr("required", "required").attr("skipped", "false");
-                    $(".bd-optional").hide();
-                } else {
-                    if (theVal === "partner") { // If partner, skip all questions
-                        $("#clinicalContainer").attr("skipped", "true");
-                        $("#orgsContainer").attr("skipped", "true");
-                        $("#date").removeAttr("required").attr("skipped", "true");
-                        $("#month").removeAttr("required").attr("skipped", "true");
-                        $("#year").removeAttr("required").attr("skipped", "true");
-                        $(".bd-optional").show();
-                    }
-                }
-            });
+        $("input[name='user_type']").on("click", function() {
+            var roles = [], theVal = $(this).val();
+            roles.push({name: theVal});
+            tnthAjax.putRoles(self.userId,{"roles": roles}, $("#rolesGroup"));
+            self.handlePostEvent(self.getSectionContainerId($(this)));
         });
     };
 
-    FieldsChecker.prototype.patientQEvent = function(fields) {
+    FieldsChecker.prototype.patientQEvent = function() {
```

Comment:
Could these validation functions be generalized for specific element types, or reused in some way?

Reply:
@ivan-c think so, ivan..although i think some of these behaviors are specific enough to warrant custom validation functions.  let me think this through a bit, will do a PR when I have a better idea how to change these.